providers/edge/src/airflow/providers/edge/cli/edge_command.py

```diff
@@ -395,9 +397,11 @@ def heartbeat(self) -> None:
                 if _EdgeWorkerCli.maintenance_mode:
                     logger.info("Exit Maintenance mode requested!")
                 _EdgeWorkerCli.maintenance_mode = False
+            worker_state_changed = worker_info.state == state
```

Comment:
Are you sure this is the _right_ check? Why has the state changed when the server response equals the reported state?